### ISSUE TITLE
java.nio.channels.ClosedChannelException when calling dispose()

### ISSUE THREAD

**Issue by piotr-signalfx** [rsocket/rsocket-java#869]
I can see java.nio.channels.ClosedChannelException logged when I run the attaching test case. 
I could not find a good example how to call dispose() properly in this case or why the error is reported here. Most examples just call block() on Mono<RSocket> and then dispose() on the returned RSocket instances. I would like to avoid calling block().

## Expected Behavior
I would like to be able to call dispose() on RSocket instance at some point in doFinally or similar callback without any exceptions/errors.

## Actual Behavior
java.nio.channels.ClosedChannelException seen in onErrorDropped.

## Steps to Reproduce
Run the following test case

```java
import java.util.concurrent.CountDownLatch;

import org.junit.Test;

import io.rsocket.Payload;
import io.rsocket.RSocket;
import io.rsocket.SocketAcceptor;
import io.rsocket.core.RSocketConnector;
import io.rsocket.core.RSocketServer;
import io.rsocket.transport.netty.client.TcpClientTransport;
import io.rsocket.transport.netty.server.TcpServerTransport;
import io.rsocket.util.DefaultPayload;
import reactor.core.publisher.Hooks;
import reactor.core.publisher.Mono;

public final class HelloWorldClient2 {

    private final static CountDownLatch latch = new CountDownLatch(1);

    @Test
    public void reproCase() throws InterruptedException {

        Hooks.onErrorDropped(e -> System.err.println(e));

        RSocket rsocket = new RSocket() {
            @Override
            public Mono<Payload> requestResponse(Payload p) {
                return Mono.just(p);
            }
        };

        RSocketServer.create(SocketAcceptor.with(rsocket)).bind(TcpServerTransport.create("localhost", 7000)).subscribe();
        RSocketConnector.connectWith(TcpClientTransport.create("localhost", 7000)).flatMap(socket -> exchange(socket)).block();
        latch.await();
    }

    private Mono<String> exchange(RSocket socket) {
        return socket.requestResponse(DefaultPayload.create("Hello"))
                .map(Payload::getDataUtf8)
                .doOnNext(System.out::println)
                .doFinally(__ -> latch.countDown())
                .doFinally(__ -> socket.dispose());
    }
}
```

* RSocket version(s) used: 1.0.0

**Comment by OlegDokuka** [comment]
@piotr-signalfx what is the purpose of calling the `dispose` method in such a case?

**Comment by piotr-signalfx** [comment]
Isn't that necessary to release the underlying connection and clean up?
Will RSocket take care of the cleanup automatically in such a scenario?

**Comment by OlegDokuka** [comment]
I'm asking since usually rsocket is not used for a single request but usually for multiplexing of requests sequence during a long connection session.

**Comment by piotr-signalfx** [comment]
Yes, the majority of scenarios involve long sessions in our case as well. However, this one is an exception where we just needed to establish a new connection and get a single response and run into the above-mentioned issue. 
We can consider using block() with dispose() as we actually do for long running sessions if needed for the time being. Anyway, it was a bit unexpected to see such an exception in the logs.

**Comment by OlegDokuka** [comment]
Alright. I know where the problem I just did not want to fix it since this is particular reactor behavior related to this kind of bug -> https://github.com/reactor/reactor-core/issues/1431.

However, I guess I can work around it

**Comment by OlegDokuka** [comment]
this might have been fixed by #859. Need to double-check whether it is true

**Comment by OlegDokuka** [comment]
With the test, we can not reproduce it in 1.1 but still see it appearing from time to time in 1.0.x. Probably, it is not worth fixing in 1.0.x since it does not affect any core functionality and rsocket disposal happens not that frequently.
